src/EpiEasyEvents/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Forte.EpiEasyEvents;
+using Microsoft.AspNetCore.Builder;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddEpiEasyEvents(this IServiceCollection services, IEnumerable<Assembly> assemblies)
+        {
+            services.AddSingleton<EventsRegistry>();
+            RegisterEventHandlers(services, assemblies);
+
+            return services;
+        }
+
+        public static IServiceCollection AddEpiEasyEvents(this IServiceCollection services, Assembly assembly)
+        {
+            return services.AddEpiEasyEvents(new[] {assembly});
+        }
+
+        public static void UseEpiEasyEvents(this IApplicationBuilder app)
+        {
+            var easyEventsRegistry = app.ApplicationServices.GetRequiredService<EventsRegistry>();
+            easyEventsRegistry.RegisterEvents();
+        }
+
+        private static void RegisterEventHandlers(IServiceCollection services, IEnumerable<Assembly> assemblies)
+        {
+            var typesToRegister = assemblies.SelectMany(a => a.GetTypes())
+                .Where(type => type.GetInterfaces().Any(x => x.IsGenericType && x.GetGenericTypeDefinition() == typeof(IContentEventHandler<,>)));
+
+            foreach (var typeToRegister in typesToRegister)
+            {
+                // Register against all the interfaces implemented
+                // by this concrete class
+                var interfacesToRegister = typeToRegister.GetInterfaces();
```

Comment:
I'm wondering if we should really register it against ALL interfaces instead of only IContentEventHandlers.
I can imagine rare scenarios when the user adds some custom interface to their IContentEventHandler but temporarily wants to remove its registration from the service provider. They can do this in their Startup but we will register it here implicitly thus possibly causing some hard to debug issues in users' applications. What do you think?

Reply:
Fixed in 4f88405e97a5a8f02bb1bfc62c02ba033432d786